tests/integration/flag_check/tester.cpp

```diff
@@ -0,0 +1,116 @@
+ * specific error occurring.
+ */
+int main(int argc, char **argv) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  memgraph::communication::SSLInit sslInit;
+
+  memgraph::io::network::Endpoint endpoint(memgraph::io::network::ResolveHostname(FLAGS_address), FLAGS_port);
+
+  memgraph::communication::ClientContext context(FLAGS_use_ssl);
+  memgraph::communication::bolt::Client client(context);
+
+  try {
+    client.Connect(endpoint, FLAGS_username, FLAGS_password);
+  } catch (const memgraph::communication::bolt::ClientFatalException &e) {
+    if (FLAGS_should_fail) {
+      if (!FLAGS_failure_message.empty() && e.what() != FLAGS_failure_message) {
+        LOG_FATAL(
+            "The query should have failed with an error message of '{}'' but "
+            "instead it failed with '{}'",
+            FLAGS_failure_message, e.what());
+      }
+      return 0;
+    } else {
+      LOG_FATAL(
+          "The query shoudn't have failed but it failed with an "
+          "error message '{}'",
+          e.what());
+    }
+  } catch (const memgraph::communication::bolt::ServerCommunicationException &e) {
+    if (FLAGS_should_fail) {
+      if (!FLAGS_failure_message.empty() && e.what() != FLAGS_failure_message) {
+        LOG_FATAL(
+            "The query should have failed with an error message of '{}'' but "
+            "instead it failed with '{}'",
+            FLAGS_failure_message, e.what());
+      }
+      return 0;
+    } else {
+      LOG_FATAL(
+          "The query shoudn't have failed but it failed with an "
+          "error message '{}'",
+          e.what());
+    }
+  }
+
+  for (int i = 1; i < argc; ++i) {
+    std::string query(argv[i]);
+    try {
+      client.Execute(query, {});
+    } catch (const memgraph::communication::bolt::ClientQueryException &e) {
+      if (!FLAGS_check_failure) {
+        if (!FLAGS_failure_message.empty() && e.what() == FLAGS_failure_message) {
+          LOG_FATAL(
+              "The query should have succeeded or failed with an error "
+              "message that isn't equal to '{}' but it failed with that error "
+              "message",
+              FLAGS_failure_message);
+        }
+        continue;
+      }
+      if (FLAGS_should_fail) {
+        if (!FLAGS_failure_message.empty() && e.what() != FLAGS_failure_message) {
+          LOG_FATAL(
+              "The query should have failed with an error message of '{}'' but "
+              "instead it failed with '{}'",
+              FLAGS_failure_message, e.what());
+        }
+        return 0;
+      } else {
+        LOG_FATAL(
+            "The query shoudn't have failed but it failed with an "
+            "error message '{}'",
+            e.what());
+      }
+    }
+    if (!FLAGS_check_failure) continue;
+    if (FLAGS_should_fail) {
+      LOG_FATAL(
+          "The query should have failed but instead it executed "
+          "successfully!");
+    }
+  }
```

Comment:
second time I see this, extract the common logic into a function and call it here and above.